front/migrations/20240829_backfill_keys_without_group_id.ts

```diff
@@ -0,0 +1,79 @@
+import type { LightWorkspaceType } from "@dust-tt/types";
+
+import { Authenticator } from "@app/lib/auth";
+import { Workspace } from "@app/lib/models/workspace";
+import { GroupResource } from "@app/lib/resources/group_resource";
+import { KeyResource } from "@app/lib/resources/key_resource";
+import { KeyModel } from "@app/lib/resources/storage/models/keys";
+import { renderLightWorkspaceType } from "@app/lib/workspace";
+import type { Logger } from "@app/logger/logger";
+import { makeScript } from "@app/scripts/helpers";
+
+async function backfillApiKeys(
+  workspace: LightWorkspaceType,
+  logger: Logger,
+  execute: boolean
+) {
+  logger.info("Handle workspace " + workspace.id);
+  const auth = await Authenticator.internalAdminForWorkspace(workspace.sId);
+
+  if (execute) {
+    const globalGroup = await GroupResource.fetchWorkspaceGlobalGroup(auth);
+    if (globalGroup.isOk()) {
+      await KeyResource.model.update(
+        { groupId: globalGroup.value.id },
+        {
+          where: {
+            workspaceId: workspace.id,
+            isSystem: false,
+          },
+        }
+      );
+    }
+
+    const systemGroup = await GroupResource.fetchWorkspaceSystemGroup(auth);
+    if (systemGroup.isOk()) {
+      await KeyResource.model.update(
+        { groupId: systemGroup.value.id },
+        {
+          where: {
+            workspaceId: workspace.id,
+            isSystem: true,
+          },
+        }
+      );
+    }
+  }
+  logger.info("Done with workspace " + workspace.id);
+}
+
+makeScript({}, async ({ execute }, logger) => {
+  const keys = await KeyModel.findAll({
+    // @ts-expect-error groupId is now not nullable in our database schema, even though we have some null values in the database.
+    where: {
+      isSystem: true,
```

Comment:
Do we have keys with missing `groupId` that are not system keys?

Reply:
No